### ISSUE TITLE
Persist mopidy state between runs.

### ISSUE THREAD

**Issue by adamcik** [mopidy/mopidy#310]
Basically adding support for persisting the current tracklist and position in it in such a way we can reload it at startup. Idea came out of now closed pull request #181

**Comment by adamcik** [comment]
@jodal didn't we talk quite a bit about this last time we chated? Did this end up in some doc or did we loose that state?

**Comment by jodal** [comment]
We recorded part of that in the the core-cleanup doc, as a part of the new `tracklist.get_state()` method:

> core.tracklist.get_state() -> dict
Get a dict with the current state of consume, random, repeat, single, version, current TLID, and maybe current index?
Include has_{eot,next,previous}_track too? Ref #1059
Useful for implementing #310

To summarize what we need to store for each of the the five core controllers:

- library: no state in core, all in backends
- playlists: no state in core, all in backends
- tracklist:
  - result of `get_tl_tracks()` (or similar in after core-cleanup)
  - result of the new `get_state()` method described above 
- playback: current_tl_track only, which is already included in `tracklist.get_state()`, so nothing more to persist here
- history: `get_history()` returns everything we need to persist

In summary, our full core state is:

``` python
{
    'tracklist': core.tracklist.get_tl_tracks(),
    'state': core.tracklist.get_state(),
    'history': core.history.get_history()',
}
```

**Comment by adamcik** [comment]
As for history, there was some talk of throwing it out of core and making it an optional "core extension" #1100 

As for the tracklist state, we would only need the `tlid` of the current `tl_track` not the whole thing.

For `get_state()` that has nothing to do with playback state, just clarify as we have to `get_state()` methods in play now.

Do we want to persist the playback state?

**Comment by adamcik** [comment]
https://github.com/bardoloi/mopidy-qsaver seems to be workaround for this.

**Comment by bardoloi** [comment]
Quick housekeeping note to @adamcik 's last comment: we've moved the repository to a different org: https://github.com/HeadspringLabs/mopidy-qsaver

..and are actively working on the suggestions here.

**Comment by adamcik** [comment]
We should also consider having the mixer volume setting `previous` and have this persist the volume as requested in #1197

**Comment by kingosticks** [comment]
There is a possible use case in pimusicbox where we'd like to save Mopidy's current state including `playback.get_state` (and maybe even `get_time_position`) too. An API to save and restore the state would be nice, rather than having it tied to startup and shutdown of Mopidy.

**Comment by languitar** [comment]
:+1:

**Comment by blueyed** [comment]
Just for reference: mpdtoys contains the `mpstore` and `mpload` tools: https://joeyh.name/code/mpdtoys/
